connection-limit/connection_limit_user.c

```diff
@@ -419,156 +375,76 @@ static void parse_tcp(char *file, int (*proc)(int, char*))
-    }
-    map_fd[2] = bpf_map__fd(bpf_object__find_map_by_name(obj, tcp_conns_map_name));
-    if (map_fd[1] < 0)
-    {
-        fprintf(stderr, "Failed to get max conn map FD\n");
-        return 1;
-    }
-    map_fd[3] = bpf_map__fd(bpf_object__find_map_by_name(obj, conn_info_map_name));
-    if (map_fd[1] < 0)
-    {
-        fprintf(stderr, "Failed to get max conn map FD\n");
-        return 1;
-    }
-    map_fd[4] = bpf_map__fd(bpf_object__find_map_by_name(obj, recv_count_map_name));
-    if (map_fd[4] < 0)
-    {
-        fprintf(stderr, "Failed to get receive count map FD\n");
-        return 1;
-    }
-    map_fd[5] = bpf_map__fd(bpf_object__find_map_by_name(obj, drop_count_map_name));
-    if (map_fd[5] < 0)
-    {
-        fprintf(stderr, "Failed to get drop count map FD\n");
-        return 1;
-    }
-    map_fd[6] = bpf_map__fd(bpf_object__find_map_by_name(obj, ingress_next_prog_map_name));
-    if (map_fd[6] < 0)
-    {
-        fprintf(stderr, "Failed to get ingress next program FD\n");
-        return 1;
-    }
-    return 0;
-}
-
-
-static struct bpf_object* load_bpf_programs(const char *bpf_file, int prog_fd[2])
+static bool validate_ifname(const char *input_ifname, char *output_ifname)
 {
-    struct bpf_object *bpf_obj;
-    struct bpf_program *bpf_prog, *bpf_prog1;
-
-    struct bpf_object_open_attr open_attr = {
-            .file = bpf_file,
-            .prog_type = BPF_PROG_TYPE_UNSPEC,
-    };
-    bpf_obj = bpf_object__open_xattr(&open_attr);
-    if (!bpf_obj)
-    {
-        fprintf(stderr, "ERR: failed to open object %s\n", bpf_file);
-        return NULL;
-    }
+    size_t len;
+    int i;
 
-    if (bpf_object__load(bpf_obj))
-    {
-        fprintf(stderr, "Failed to load BPF Object\n");
-        return NULL;
-    }
-    bpf_prog = bpf_program__next(NULL, bpf_obj);
-    if (!bpf_prog)
-    {
-        fprintf(stderr, "Couldn't find a program trace_inet_sock_set_state in the BPF file %s\n", bpf_file);
-        return NULL;
+    len = get_length((void *)input_ifname);
+    if (len >= IF_NAMESIZE) {
+        return false;
     }
+    for (i = 0; i < len; i++) {
+        char c = input_ifname[i];
 
-    prog_fd[0] = bpf_program__fd(bpf_prog);
-    if (prog_fd[0] <= 0) {
-        fprintf(stderr, "Failed to get program FD\n");
-        return NULL;
+        if (!(isalpha(c) || isdigit(c))) {
+            return false;
+        }
     }
+    cpy(output_ifname, (void *)input_ifname);
+    return true;
+}
 
-    bpf_prog1 = bpf_program__next(bpf_prog, bpf_obj);
-    if (!bpf_prog1)
-    {
-        fprintf(stderr, "Couldn't find a program _xdp_limit_conn in the BPF file %s\n", bpf_file);
-        return NULL;
-    }
-    prog_fd[1] = bpf_program__fd(bpf_prog1);
-    if (prog_fd[1] <= 0) {
-        fprintf(stderr, "Failed to get program FD\n");
-        return NULL;
+void cpy(char *dst, char *src)
+{
+    while ((*src) != '\0') {
+        *dst++ = *src++;
     }
-
-    return bpf_obj;
+    *dst = '\0';
 }
```

Comment:
why can't you use strcpy()?

Reply:
We had to implement a custom copy method in order to resolve the code checker errors.

Reply:
What code checker errors?